Sources/MySQLKit/MySQLConfiguration.swift

```diff
@@ -31,9 +31,7 @@ public struct MySQLConfiguration {
         guard let hostname = url.host else {
             return nil
         }
-        guard let port = url.port else {
-            return nil
-        }
+        let port = url.port ?? 3306
```

Comment:
I'm definitely in favor of this change, but I don't like hardcoding the default port again when the `port` parameter of `self.init` already defaults to it. Unfortunately, making use of that defaulted parameter would require an invocation something like this:

```swift
if let port = url.port {
    self.init(hostname: hostname, port: port, username: username, password: password, database: url.path.split(separator: "/").last.map(String.init), tlsConfiguration: tlsConfiguration)
} else {
    self.init(hostname: hostname, username: username, password: password, database: url.path.split(separator: "/").last.map(String.init), tlsConfiguration: tlsConfiguration)
}
```

We can't change the initializer's signature to solve the problem because it would cause a public API break (semver major).

My suggestion is to add this after line 10:

```swift
    /// IANA-assigned port number for MySQL
    /// `UInt16(getservbyname("mysql", "tcp").pointee.s_port).byteSwapped`
    public static var ianaPortNumber: Int { 3306 }
```
And then replace the `3306` in _both_ initializers with `Self.ianaPortNumber`.

Worth noting `postgres-kit` would benefit from an similar edit.

Reply:
(My suggestion does involve technically adding new public API in the form of the static property, which would make the PR semver minor instead of semver patch, but that's not a big deal. After all, technically speaking, changing the existing behavior of the URL-based initializer should be considered semver major 🙂.)

Reply:
@gwynne Updated